x-pack/plugins/triggers_actions_ui/public/application/sections/rule_details/components/rule_actions.tsx

```diff
@@ -37,31 +45,44 @@ export function RuleActions({ ruleActions, actionTypeRegistry }: RuleActionsProp
         </EuiText>
       </EuiFlexItem>
     );
+  }
 
-  function getActionIconClass(actionGroupId?: string): IconType | undefined {
+  const getActionIconClass = (actionGroupId?: string): IconType | undefined => {
     const actionGroup = actionTypeRegistry.list().find((group) => group.id === actionGroupId);
     return typeof actionGroup?.iconClass === 'string'
       ? actionGroup?.iconClass
       : suspendedComponentWithProps(actionGroup?.iconClass as React.ComponentType);
-  }
+  };
+
+  const getActionName = (actionTypeId?: string) => {
+    const actionConnector = actionConnectors.find((connector) => connector.id === actionTypeId);
+    return actionConnector && actionConnector.name;
+  };
+
   if (isLoadingActionConnectors) return <EuiLoadingSpinner size="s" />;
   return (
     <EuiFlexGroup direction="column" gutterSize="none">
-      {actionConnectors.map(({ actionTypeId, name }) => (
-        <EuiFlexItem key={actionTypeId}>
-          <EuiFlexGroup alignItems="center" gutterSize="s" component="span">
-            <EuiFlexItem grow={false}>
-              <EuiIcon size="m" type={getActionIconClass(actionTypeId) ?? 'apps'} />
-            </EuiFlexItem>
-            <EuiFlexItem>
-              <EuiText data-test-subj={`actionConnectorName-${name}`} size="s">
-                {name}
-              </EuiText>
-            </EuiFlexItem>
-          </EuiFlexGroup>
-          <EuiSpacer size="s" />
-        </EuiFlexItem>
-      ))}
+      {ruleActions.map(({ actionTypeId, id }, index) => {
```

Comment:
So previously we iterated on the connectors that the rule is using for its actions, so say the rule has 4 actions:

- slack
- slack
- email
- server log

these 4 actions only use 3 connectors since the slack connector is re-used. Therefore, iterating on the connectors would give us 3 entities. Iterating on the actions will give 4.

The bug that this PR fixes is we were not showing all of the actions that a rule has, hence the change to iterating on `ruleActions`.